api/host.py

```diff
@@ -365,7 +365,10 @@ def update_facts_by_namespace(operation, host_id_list, namespace, fact_dict):
         else:
             host.merge_facts_in_namespace(namespace, fact_dict)
 
-    db.session.commit()
+    if db.session.is_modified(host):
+        db.session.commit()
+        serialized_host = serialize_host(host, staleness_timestamps(), EGRESS_HOST_FIELDS)
+        _emit_patch_event(serialized_host, host.id, host.canonical_facts.get("insights_id"))
```

Comment:
This needs to be inside the `for host` loop in order to produce an update for each host:
```suggestion
        if db.session.is_modified(host):
            db.session.commit()
            serialized_host = serialize_host(host, staleness_timestamps(), EGRESS_HOST_FIELDS)
            _emit_patch_event(serialized_host, host.id, host.canonical_facts.get("insights_id"))
```